agreement_legal/models/agreement.py

```diff
@@ -0,0 +1,458 @@
+# Copyright (C) 2018 - TODAY, Pavlov Media
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class Agreement(models.Model):
+    _name = "agreement"
+    _inherit = ["agreement", "mail.thread"]
+
+    def _default_stage_id(self):
+        return self.env.ref("agreement_legal.agreement_stage_new")
+    # re-adding company from "agreement"
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        default=lambda self: self.env["res.company"]._company_default_get("agreement"),
+    )
```

Comment:
@ygol Should be removed. No need to redefine it. The inheritance should do its job here.